File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Arms/AlibabaCloud/SDK/ROS/CDK/Arms/Datasource/PrometheusAuthTokenProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Arms.Datasource
{
    /// <summary>Properties for defining a `PrometheusAuthToken`.</summary>
    /// <remarks>
    /// See https://www.alibabacloud.com/help/ros/developer-reference/datasource-arms-prometheusauthtoken
    /// </remarks>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-arms.datasource.PrometheusAuthTokenProps")]
    public class PrometheusAuthTokenProps : AlibabaCloud.SDK.ROS.CDK.Arms.Datasource.IPrometheusAuthTokenProps
    {
        private object? _clusterId;

        /// <summary>Property clusterId: Cluster ID.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "clusterId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? ClusterId
        {
            get => _clusterId;
            set
            {
                if (Amazon.JSII.Runtime.Configuration.RuntimeTypeChecking)
                {
                    switch (value)
                    {
                        case string cast_cd4240:
                            break;
                        case AlibabaCloud.SDK.ROS.CDK.Core.IResolvable cast_cd4240:
                            break;
                        case Amazon.JSII.Runtime.Deputy.AnonymousObject cast_cd4240:
                            // Not enough information to type-check...
                            break;
                        case null:
                            break;
                        default:
                            throw new System.ArgumentException($"Expected {nameof(value)} to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IResolvable).FullName}; received {value.GetType().FullName}", nameof(value));
                    }
                }
                _clusterId = value;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkarms/datasource/PrometheusAuthTokenProps.go
package datasource


// Properties for defining a `PrometheusAuthToken`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/datasource-arms-prometheusauthtoken
type PrometheusAuthTokenProps struct {
	// Property clusterId: Cluster ID.
	ClusterId interface{} `field:"optional" json:"clusterId" yaml:"clusterId"`
	// Property refreshOptions: The refresh strategy for the datasource resource when the stack is updated.
	//
	// Valid values:
	// - Never: Never refresh the datasource resource when the stack is updated.
	// - Always: Always refresh the datasource resource when the stack is updated.
	// Default is Never.
	RefreshOptions interface{} `field:"optional" json:"refreshOptions" yaml:"refreshOptions"`
}

